server/routes/viewLocationsRouter.ts

```diff
@@ -34,6 +34,13 @@ export const addActions = asyncMiddleware(async (req, res, next) => {
     })(req, res, null)
   }
 
+  if (req.canAccess('change_non_residential_type') && !isResidential && leafLevel) {
```

Comment:
Due to the way that `leafLevel` is defined in the API, it will return true for any location that doesn't have sublocations, including wings/landings/etc. I think maybe we want to check `locationType === 'ROOM'` here instead?